articles/data-leakage/index.md

```diff
@@ -0,0 +1,83 @@
+
+#### Same Data for Testing and Validation
+
+It is standard to split data into separate groups such as testing, training, and validation sets. Not all models require a validation set since we can train a model using the training set and use the test set to test it. As standard as this process is, it becomes a cause of data leakage if not done correctly. For example, if we unintentionally expose the model to test data during training, the model will already have access to data, it is not yet allowed to see. This then allows a model to perform better in its predictions regarding the test data. A problem associated with this performance is that during deployment, the same model performs underwhelmingly. Data leakage impedes models from generalizing well after in the real-world, facing new data.
+
+#### Duplicates
+
+Duplication is present if a dataset has many data points that have the same or almost identical data. This is a common occurrence when dealing with real-world, often noisy data. If test data and training data contain the same instance, data leakage may be experienced even if representing different observations.
+
+#### Implicit Leakage
+
+Consider temporal data. This is data related to time. Time is a crucial factor. An example is time-series data. Imagine having a training set with two data points, 1 and 3. Let's have the test set with a single data point labeled 2. Let's assume that the temporal sequence of these points is 1, 2, then 3. There is a likelihood that we have introduced data leakage thanks to how we created our test and training sets. Since point 3 is in the training set, and 2 is in the test set. In this situation, we are unrealistically training our model on future information compared to the position of the test data in time. We inadvertently end up leaking information about the future into the past.
+
+### Techniques to Minimize Leakage
+
+Since we now know the possible causes of leakage, we should explore the various ways to minimize or prevent data leakage. Let's go through three effective methods.
+
+#### Normalizing Correctly Before Cross-Validation
+
+It is easy to leak data when preparing data for machine learning. As a result, overfitting occurs in training data. This also impacts the performance of the model on test data. We would be given a flattering picture of performance, but it performs poorly when the model is deployed in the real world.
+For instance, data leakage will occur in a scenario where one carries out normalization of an entire dataset, then evaluates the algorithm's performance using cross-validation. [Normalization](https://developers.google.com/machine-learning/data-prep/transform/normalization#:~:text=%20Normalization%20Techniques%20at%20a%20Glance%20%201,a%20wide%20range%20to%20a%20narrow...%20More%20) is a process that aims to improve the performance of a model by transforming features to be on a similar scale. Data normalization is carried out by dividing the data by its mean.
```

Comment:
There are different types of normalization. The last sentence is true only in a specific kind. There are min-max normalization, mean-subtraction, etc. Remove the last sentence.